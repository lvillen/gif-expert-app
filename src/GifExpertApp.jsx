import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['Lord of the Rings']);

    return(
        <>
            <h2>Gif Expert App</h2>
            <AddCategory setCategoriesProp={ setCategories }/>
            <hr />

            <ul>
              {categories.map( category => (
                <GifGrid 
                  key={ category }
                  categoryProp={ category }
                />
              ))
              }
            </ul>
        </>
    );
};
