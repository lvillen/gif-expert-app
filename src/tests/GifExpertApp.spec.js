import React from 'react';
import { GifExpertApp } from '../GifExpertApp';
import { shallow } from 'enzyme';

describe('Tests GifExpertApp Component', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<GifExpertApp/>);

    expect( wrapper ).toMatchSnapshot();
  });

  it('should show a list of categories', () => {
    const categories = ['Lord of the Rings', 'Game of Thrones'];
    const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('GifGrid').length ).toBe( categories.length );
  });
});