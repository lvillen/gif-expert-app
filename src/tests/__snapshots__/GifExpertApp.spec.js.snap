// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests GifExpertApp Component should match the snapshot 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategoriesProp={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;

exports[`Tests GifExpertApp Component should show a list of categories 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategoriesProp={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      categoryProp="Lord of the Rings"
      key="Lord of the Rings"
    />
    <GifGrid
      categoryProp="Game of Thrones"
      key="Game of Thrones"
    />
  </ul>
</Fragment>
`;
